### PR TITLE
Add a chatbox and some misc groundwork

### PR DESCRIPTION
**Pull request by TheSecondReal0**
Added a chatbox that lets players talk to each other ingame. The chatbox can display messages on opposite sides, handle names and colors, do smart autoscrolling, and scales well to any size. Also added 2 new autoload scripts to keep track of data and to help sync the gamestate, GameManager (gamemanager.gd) and PlayerManager (playermanager.gd). Also, the player can no longer move around while in a menu.

### CODE REVIEW

**Comment by tooManyArguments** [comment]
please rebase again and check on conflicts

**Comment by TheSecondReal0** [comment]
Conflicts have been fixed

File: src/assets/player/player.gd [hunk 1]
```diff
@@ -22,16 +22,16 @@ func _ready():
...
+	velocity = Vector2(0, 0)
+	if not PlayerManager.inMenu:
+		if Input.is_action_pressed('ui_right'):
+			velocity.x = 1
+		if Input.is_action_pressed('ui_left'):
+			velocity.x = -1
+		if Input.is_action_pressed('ui_down'):
+			velocity.y = 1
+		if Input.is_action_pressed('ui_up'):
+			velocity.y = -1
```
Comment (vainiovano) [inline src/assets/player/player.gd]: Replace `velocity` with `movement` and don't remove the normalization

**Review by monban — APPROVED** [review]
Code looks solid, tested build, game still runs, no missing features. Tested chat, seems to be working as intended.

**Pull request merged**